### PR TITLE
feat: enable secured grpc custom gateway

### LINKED ISSUE #1008: feat: expose /tags and /fields endpoint from API
The FE needs to be able to grab information such as `/tags` and `/fields` (index fields & encoders) from the API. Currently, the playground just makes calls directly to the flow with the client. Instead, we should define these endpoints as part of the BFF.

### LINKED ISSUE #1034: feat: integrate grpc into custom gateway
The associated PR
- enables grpc support (closes #1034)
  - avoids general size limitations from http & nginx (was also solved in https://github.com/jina-ai/now/issues/1032) as we are using grpc again
- note the custom gateway is typically started on `port=[8081, 8085` and `protocol=['http', 'grpc']` (there is some freedom in choosing those values but probably easiest to stick to those values; more infos in the docs of the class)
- circumvents the issue that `StreamlitServer` doesn't respect the passed command line arguments by writing to streamlit config file
  - solves the issue that playground reloads from `...wolf.jina.ai/playground` to `...wolf.jina.ai`
  - solves the issue that playground immediately starts when starting custom gateway
- in order to pass `UserInput` to the BFF & playground, we need to write it to disk and read it (I chose the home directory)
  - issue with playground: `StreamlitServer` doesn't respect command line arguments
  - issue with FastAPI: it's unclear how to pass CLI arguments through uvicorn in this setup and we load the installed `now` package when running the BFF
  - removes calling the flow for `UserInput` attribute in the playground
  - closes #1008 by exposing the frontend utils under `/api/v1/frontend-utils` and removes relevant logic from indexer
- fixes securitisation of playground
  - upgrades fetching of cookies in streamlit to used version
- removes unused code in the playground
- removes unused request model parameters in the bff

### PR DESCRIPTION
**Pull request by joschkabraun**
This PR
- enables grpc support (closes #1034)
  - avoids general size limitations from http & nginx (was also solved in https://github.com/jina-ai/now/issues/1032) as we are using grpc again
- note the custom gateway is typically started on `port=[8081, 8085` and `protocol=['http', 'grpc']` (there is some freedom in choosing those values but probably easiest to stick to those values; more infos in the docs of the class)
- circumvents the issue that `StreamlitServer` doesn't respect the passed command line arguments by writing to streamlit config file
  - solves the issue that playground reloads from `...wolf.jina.ai/playground` to `...wolf.jina.ai`
  - solves the issue that playground immediately starts when starting custom gateway
- in order to pass `UserInput` to the BFF & playground, we need to write it to disk and read it (I chose the home directory)
  - issue with playground: `StreamlitServer` doesn't respect command line arguments
  - issue with FastAPI: it's unclear how to pass CLI arguments through uvicorn in this setup and we load the installed `now` package when running the BFF
  - removes calling the flow for `UserInput` attribute in the playground
  - closes #1008 by exposing the frontend utils under `/api/v1/frontend-utils` and removes relevant logic from indexer
- fixes securitisation of playground
  - upgrades fetching of cookies in streamlit to used version
- removes unused code in the playground
- removes unused request model parameters in the bff

Customer deployment of secured playground (see logged in account)

Screenshot 2023-02-10 at 09 38 58

Search Results:

Screenshot 2023-02-10 at 09 39 05

We need to login:

Screenshot 2023-02-10 at 10 41 03

Screenshot 2023-02-10 at 10 41 12

Screenshot 2023-02-10 at 10 41 16

Screenshot 2023-02-10 at 10 44 04

---

- [ ] This PR references an open issue
- [ ] Added a test
- [ ] Updated the documentation
- [ ] Added a screenshot of a successful customer deployment

### CODE REVIEW

File: now/executor/gateway/bff/app/settings.py [hunk 1]
```diff
@@ -34,3 +37,19 @@
     },
     'root': {'level': 'DEBUG', 'handlers': ['console']},
 }
+
+
+user_input_in_bff = UserInput()
+try:
```
Comment (florian-hoenicke) [inline now/executor/gateway/bff/app/settings.py:43]: The `user_input.json` file is dumped in the constructor. In what scenario won't it be there?

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

File: now/executor/gateway/bff/app/settings.py (hunk 1 above)
Comment (joschkabraun) [reply now/executor/gateway/bff/app/settings.py:43]: local testing

File: now/app/base/app_utils.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from now.constants import DEFAULT_FLOW_NAME
+
+
+def create_jcloud_name(flow_name: str) -> str:
```
Comment (florian-hoenicke) [inline now/app/base/app_utils.py:4]: app_utils is a generic name. Can we find something that suits better?

File: now/executor/gateway/bff/app/app.py [hunk 3]
```diff
@@ -68,7 +68,7 @@
~
~    @app.on_event('startup')
~    def startup():
~        logger.info(
~            f'Jina NOW started! ' f'Listening to [::]:{api_settings.DEFAULT_PORT}'
```
Comment (florian-hoenicke) [inline now/executor/gateway/bff/app/app.py]: I think we could have a better naming than frontend utils. It does not necessarily have to be used by a frontend and utils does not describe the functionality.
What about `info`? We can get information about the deployment.

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

**Review by florian-hoenicke — APPROVED** [review]
Great work! With this, we have a reliable service to offer.

**Pull request merged**